device/transport/amqp/devdoc/amqp_twin_client_requirements.md

```diff
@@ -3,106 +3,139 @@
 ## Overview
 Object used to subscribe to the Cloud-to-Device messages for Twin
 
-## Example
-```javascript
-var receiver = new AmqpTwinClient(config: ClientConfig, client: any);
-receiver.on('response', function(response) {
-    console.log('Response received for request ' + response.requestId);
-    console.log('  status = " + response.status);
-    console.log(response.body);
-});
+## Public API
+
+```typescript
+class AmqpTwinClient extends EventEmitter {
```

Comment:
>AmqpTwinClient [](start = 6, length = 14)

I haven't gone back to the MQTT, but is this the same interface that MQTT uses?  If so, do we want to codify it as an interface?

Reply:
it sort of is but since these are "hidden" in the transport i'm not sure where i'd put this or what purpose it would serve. each transport might evolve independently and as you said since there is a smell associated with a separate twin client if we end up changing the pattern i would rather have decoupled transports,

---
In reply to: [171681233](https://github.com/pierreca/azure-iot-sdk-node/pull/1#discussion_r171681233) [](ancestors = 171681233)